### DISCUSSION TITLE
This library is very large

### DISCUSSION THREAD

**Discussion by Paul6552** [add2cal/add-to-calendar-button#576] [Ideas / Requests]
Experience has shown that the user experience depends on the bundle size. The longer it takes to launch a website, the more frustrating the user experience is.

However, when I look at my app, I see that this library leaves a very large footprint.

Unfortunately I don't know how, but somehow it must be possible to make this library smaller.

![image](https://github.com/add2cal/add-to-calendar-button/assets/25934591/8a63a33d-e4cd-4f34-bc4a-ae573dc48094)

**Comment by jekuer** [comment]
yes and no.

Generally speaking, offering a flexible web component with all batteries included, where you can change configuration at runtime, comes at the cost of some size.
We would need to drop features in order to get it cleaner again.
For the future, we also want to make it adapt better to major JS frameworks to at least have maybe some tree shaking possibility in those cases. But there is not even a concept yet.

For now, we offer an unstyle version, where we skip the inclusion of the css, which brings it down by ~30%.
Unfortunately, this is not yet transferred to the React Wrapper. If you are using this way, I would recommend to either switch to the main package or load the JS async from the jsdelivr CDN.

**Comment by Paul6552** [comment]
Ok, so the “Unstyled Version” at least sounds promising. 30% is not bad. I am a developer myself who has only integrated one function from this library (without styling).

Yes, as already assumed, I use React. I'll try to integrate the main package without the React wrapper.

Maybe it's also smart to extract the languages. For example, I see support for many languages ​​that I don't even support in my application. English by default and if you want you can add the other languages ​​as an external file.

I18n supports external files for language support:
[i18n add language after init support](https://www.i18next.com/how-to/add-or-load-translations#add-after-init)

**Reply by jekuer** [reply]
Yes. Languages would be also a good candidate to bring in optimization here. At least for the npm package.
However, it is quite small in the end.

We also do not minimize the code as this is usually better done when building your app. But it would show up differently in your graphic - maybe we should revert it 😉

**Reply by Paul6552** [reply]
I've already tried the unstyled version and it's a lot smaller. Very good. Maybe you should advertise this on the website that such a version also exists.

Yes, the languages ​​would only bring a small saving.

And the library size image wouldn't change since this is already the bundled webpack optimized version.

Only as side information:
The unstyled version is reduced by 39%.

**Comment by khamaileon** [comment]
Where can I find the "unstyle" version?

**Reply by jekuer** [reply]
https://add-to-calendar-button.com/advanced-use#case-11
